### PR TITLE
Docs: Introduce Docusaurus website

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
#160 introduced quite a lot of documentation which lead me to do a little investigation into [docusaurus](https://docusaurus.io/). 

Screenshot 2022-12-06 at 10 35 00

Having invested a couple of hours in moving documentation around and giving this some sort of lick of paint I think it's worth pursuing with the intention of hosting on GH pages for the time being. 

Checkout the branch and run `yarn && yarn docs` to take it for a spin.

Potentially the beginning of a developer portal? 🤔 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by tolzhabayev** [comment]
Well let's take it full circle? 

Looks like the `docusaurus.config.js` is already there and correct. Workflows are missing: https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

I can add the PAT to the repo.

**Review by tolzhabayev — APPROVED** [review]
Great work, some small nits.

Release the 🦖

File: docusaurus/docs/docker.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
```
[the remark this answers is not in the document]
Comment (jackw) [reply docusaurus/docs/docker.md:1]: Nope, kept all the empty page files but removed them from the sidebar. 👍

File: docusaurus/website/src/css/custom.css [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Any CSS included here will be global. The classic template
+ * bundles Infima by default. Infima is a CSS framework designed to
+ * work well for content-centric websites.
+ */
+
+/* You can override the default Infima variables here. */
+:root {
+  --ifm-color-primary: #eb7b18;
```
[the remark this answers is not in the document]
Comment (jackw) [reply docusaurus/website/src/css/custom.css:9]: I opted for the [orange and orangeDark](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/themes/createV1Theme.ts#L134-L135) from the Grafana app theme as they felt less harsh / warmer than flamingo. Should probably run it past UX at some point. 👍

File: docusaurus/website/src/components/HomepageFeatures/index.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    description: (
+      <>
+        No need to learn and configure all the build tools necessary to build a Grafana plugin. Instant plugin
+        development environment helps you focus on coding. Easily distribute your plugin with Github workflows.
+      </>
+    ),
+  },
+  {
+    title: 'Focus on What Matters',
+    description: <>What does matter though?</>,
```
[the remark this answers is not in the document]
Comment (jackw) [reply docusaurus/website/src/components/HomepageFeatures/index.tsx]: Have updated it with some blurb. 😄

**Pull request merged**